node/network/bitfield-distribution/src/lib.rs

```diff
@@ -580,7 +580,7 @@ where
 	fn start(self, ctx: C) -> SpawnedSubsystem {
 		SpawnedSubsystem {
 			name: "bitfield-distribution-subsystem",
-			future: Box::pin(async move { Self::run(self, ctx) }.map(|_| ())),
+			future: self.run(ctx).map(|_| ()).boxed(),
```

Comment:
Why does this function even returns a `Result` when we ignore it anyway?

Reply:
I believe errors should be traced in these cases

Reply:
Maybe the `Output` type of this `Future` can be changed to `Result` and logging be done in the overseer

Reply:
Either that, or probably better propagate it to the overseer or whoever polls them and log the errors at a central place.

Reply:
Agree, but that's out of scope of this PR. The only reason I touched that line at all was because subsystems whose future used the former construction appeared to not run their `run` function at all.

Reply:
https://github.com/paritytech/polkadot/issues/1846